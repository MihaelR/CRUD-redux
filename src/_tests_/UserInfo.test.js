import ReactDom from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDom.render(<button></button>, div);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<button>Go Back</button>);
  expect(getByTestId("test1")).toHaveTextContent("Go Back");
});

it("matches shapshot", () => {
  const tree = renderer.create(<button>Go Back</button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches shapshot", () => {
  const tree = renderer.create(<button>xxx</button>).toJSON();
  expect(tree).toMatchSnapshot();
});
