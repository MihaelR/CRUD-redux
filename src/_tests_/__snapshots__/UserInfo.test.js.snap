// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches shapshot 1`] = `
<button>
  Go Back
</button>
`;

exports[`matches shapshot 2`] = `
<button>
  xxx
</button>
`;
